hidi/factorization.py

```diff
@@ -0,0 +1,199 @@
+import warnings
+import numpy as np
+import collections
+from numpy.random import permutation
+from sklearn.decomposition import TruncatedSVD
+from hidi.transform import Transform
+
+# Catch annoying warnings from nimfa
+with warnings.catch_warnings():
+    warnings.simplefilter("ignore")
+    import nimfa as nf
+
+
+class W2VStringTransform(Transform):
+    """
+    Takes a pandas Dataframe and transforms it into a
+    string
+
+    :param n_shuffles: The number of suffles for the
+    `item_id`.
+    :type n_shuffles: int
+    """
+    def __init__(self, n_shuffles=3, **w2v_kwargs):
+        self.w2v_kwargs = w2v_kwargs
+        self.n_shuffles = n_shuffles
+
+    def transform(self, df, **kwargs):
+        """
+        :param df: a pandas Dataframe with two columns:
+        :code:`link_id`, :code:`item_id`
+        :type df: pandas.Dataframe
+        :rtype: str
+        """
+        if 'item_id' not in df.index:
+            df.set_index('item_id', inplace=True)
+
+        words = ''
+        for index in df.index.unique():
+            a0 = df.loc[index].link_id
+            a = a0
```

Comment:
Can we name these variables with something more than 1-2 letters? Something that describes what they are.